src/bank_state.rs

```diff
@@ -524,7 +454,7 @@ impl BankState {
 
         let now = Instant::now();
         // Once the accounts are new transactions can enter the pipeline to process them
-        self.unlock_accounts(&txs, &results);
+        head.unlock_accounts(&txs, &results);
```

Comment:
this code needs a little rework, right?  record() above can fail and we're working on a fork inside the single bank per fullnode?  what happens to a fork that's done with a block (when record() fails)?